src/meshes/p4est_mesh_view.jl

```diff
@@ -99,23 +108,180 @@ function extract_interfaces(mesh::P4estMeshView, interfaces_parent)
     return interfaces
 end
 
+# Remove all boundaries that are not part of this p4est mesh view and add new boundaries
+# that were interfaces of the parent mesh.
+function extract_boundaries(mesh::P4estMeshView, boundaries_parent, interfaces_parent,
+                            interfaces)
+    # Remove all boundaries that are not part of this p4est mesh view.
+    boundaries = deepcopy(boundaries_parent)
+    mask = BitArray(undef, nboundaries(boundaries_parent))
+    for boundary in 1:size(boundaries_parent.neighbor_ids)[1]
+        mask[boundary] = boundaries_parent.neighbor_ids[boundary] in mesh.cell_ids
+    end
+    boundaries.neighbor_ids = global_element_id_to_local(boundaries_parent.neighbor_ids[mask],
+                                                         mesh)
+    boundaries.name = boundaries_parent.name[mask]
+    boundaries.node_indices = boundaries_parent.node_indices[mask]
+
+    # Add new boundaries that were interfaces of the parent mesh.
+    for interface in 1:size(interfaces_parent.neighbor_ids)[2]
+        if ((interfaces_parent.neighbor_ids[1, interface] in mesh.cell_ids) ⊻
+            (interfaces_parent.neighbor_ids[2, interface] in mesh.cell_ids))
+            # Determine which of the ids is part of the mesh view.
+            if interfaces_parent.neighbor_ids[1, interface] in mesh.cell_ids
+                neighbor_id = interfaces_parent.neighbor_ids[1, interface]
+                view_idx = 1
+            else
+                neighbor_id = interfaces_parent.neighbor_ids[2, interface]
+                view_idx = 2
+            end
+
+            push!(boundaries.neighbor_ids,
+                  global_element_id_to_local(neighbor_id, mesh))
+            if interfaces_parent.node_indices[view_idx, interface] == (:end, :i_forward)
+                push!(boundaries.name, :x_pos)
+            elseif interfaces_parent.node_indices[view_idx, interface] ==
+                   (:begin, :i_forward)
+                push!(boundaries.name, :x_neg)
+            elseif interfaces_parent.node_indices[view_idx, interface] ==
+                   (:i_forward, :end)
+                push!(boundaries.name, :y_pos)
+            else
+                push!(boundaries.name, :y_neg)
+            end
+
+            push!(boundaries.node_indices,
+                  interfaces_parent.node_indices[view_idx, interface])
+        end
+    end
+
+    boundaries.u = zeros(typeof(boundaries_parent.u).parameters[1],
```

Comment:
Not sure if a [`promote`](https://docs.julialang.org/en/v1/manual/conversion-and-promotion/#Promotion) is helpful for type stability